### ISSUE TITLE
Try/except exception handling index out of range errors

### ISSUE THREAD

**Issue by raghureddyram** [rmotr-curriculum/post-your-question-questions#72]
I'm not sure how to handle index out of range errors. In algorithm problems I am usually trying to practice recursion, and I'm not able to determine an easy way to handle index out of range exceptions. I would like to find a way to check for an item in a list in one line if possible. Can I do that with try/except?

**Assignment**: http://learn.rmotr.com/python/post-a-question/post-your/questions

**User code**:
```python
# your code here (if any)


# what is a pythonic way to avoid an index out of range error?
# in javascript I don't have to handle range errors. Is there an agreed upon way to prevent this in python?
# on stackoverflow I've seen try and except, but the exception either pass or returns.
# i want to say 
def test_some_call(arr, n, target):
    
    
    #in js to handle an index that doesn't exist in an array and prevent code blowing up on undefined I would write the following:
    #if(arr[n] && arr[n].length > target){//do something }
    
    #goal is:
    if arr[n] and len(arr[n]) > target:
        #do something
        
    # Is it the following?
    #1st option)
    try:
        if arr[n] and len(arr[n]) > target:
            # do something
        else:
            # do something else
    except Exception, e:
        pass
        # or is it
        return

    
    #2nd option
    try:
        if arr[n] and len(arr[n]) > target:
            # do something
            except Exception, e:
                pass
                # or is it
                return
        else:
            # do something else
   
    
# basically the indentation is throwing me off. Do I need to try/except the entire code piece? Is it best practice to try/except anytime i look up by index in a list?
# is there a way to try/except in one line?

```

**Comment by jsymons** [comment]
Instead of trying to try/except to prevent an index out of bounds (especially with a catch all exceptions block), it would be better to write your code such that it prevents the error from happening in the first place. For example:
```python
if n < len(arr) and arr[n] and len(arr[n]) > target:
    # do something
```
The `n < len(arr)` ensures that n is not outside the upper bounds of the array, if it's possible it could be a negative number you could instead use `0 <= n < len(arr)`. `arr[n]` then checks to make sure that the value at `n` isn't *Falsey*, you could also use `arr[n] is not None` to ensure it's not None, you technically don't even need this statement as the previous one ensures there **something** at n, even if that something might be `None`.
